### PR TITLE
Update release-next to 23.0

### PR DESCRIPTION
**Pull request by dannon**
With the delayed release of 22.05 we are moving straight to 23.0.

This contains just updates a few bits that were set to 22.09 during the creation of 22.05 before we knew that we'd skip.

**I'm leaving this draft as an RFC, with the question being:  should we start with 23.1 or 23.0?**

I have no preference, but my recollection is that the original leaning was that it's not semver (in process) regardless of how we cut it, so the original leaning had been to stay away from it *looking* like semver.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 VENV?=.venv
 # Source virtualenv to execute command (flake8, sphinx, twine, etc...)
 IN_VENV=if [ -f "$(VENV)/bin/activate" ]; then . "$(VENV)/bin/activate"; fi;
-RELEASE_CURR:=22.09
+RELEASE_CURR:=23.1
```
Comment (nsoranzo) [inline Makefile]:
```suggestion
RELEASE_CURR:=23.0
```

File: doc/source/releases/23.1_announce.rst [hunk 2]
```diff
@@ -2,11 +2,11 @@
 :orphan:
 
 ===========================================================
-September 2022 Galaxy Release (v 22.09)
+Early 2023 Galaxy Release (v 23.1)
```
Comment (nsoranzo) [inline doc/source/releases/23.1_announce.rst]:
```suggestion
Early 2023 Galaxy Release (v 23.0)
```

File: lib/galaxy/version.py [hunk 3]
```diff
@@ -1,3 +1,3 @@
-VERSION_MAJOR = "22.09"
+VERSION_MAJOR = "23.1"
```
Comment (nsoranzo) [inline lib/galaxy/version.py]:
```suggestion
VERSION_MAJOR = "23.0"
```

**Review by nsoranzo — APPROVED** [review]

**Comment by mvdbeek** [comment]
I think the release scripts also need updating ?

**Comment by dannon** [comment]
@mvdbeek Yep, they do.  I have a separate branch I'm working on for that -- this is just to finalize the answer to the question of 23.0 or 23.1.

**Comment by dannon** [comment]
Alright, we're going with 23.0.

**Pull request merged**